ksmt-symfpu/src/main/kotlin/org/ksmt/symfpu/SymFPUModel.kt

```diff
@@ -0,0 +1,179 @@
+package org.ksmt.symfpu
+
+import org.ksmt.KContext
+import org.ksmt.decl.KDecl
+import org.ksmt.expr.KArrayConst
+import org.ksmt.expr.KArrayLambdaBase
+import org.ksmt.expr.KArrayStoreBase
+import org.ksmt.expr.KConst
+import org.ksmt.expr.KExpr
+import org.ksmt.expr.KUninterpretedSortValue
+import org.ksmt.solver.KModel
+import org.ksmt.solver.model.KModelEvaluator
+import org.ksmt.solver.model.KModelImpl
+import org.ksmt.sort.KArraySortBase
+import org.ksmt.sort.KFpSort
+import org.ksmt.sort.KSort
+import org.ksmt.sort.KUninterpretedSort
+import org.ksmt.symfpu.ArraysTransform.Companion.mkAnyArrayLambda
+import org.ksmt.symfpu.ArraysTransform.Companion.mkAnyArraySelect
+import org.ksmt.symfpu.ArraysTransform.Companion.mkAnyArrayStore
+import org.ksmt.utils.cast
+import org.ksmt.utils.uncheckedCast
+
+class SymFPUModel(private val kModel: KModel, val ctx: KContext, val transformer: FpToBvTransformer) : KModel {
+    override val declarations: Set<KDecl<*>>
+        get() = kModel.declarations +
+            transformer.arraysTransform.mapFpToBvDeclImpl.keys -
```

Comment:
It is better to map back transformed declarations from `kModel.declarations` instead of add all declarations from the transformer. The reason is that the model may not contain some declarations. E.g., when some declarations do not affect formula satisfiability.